core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -278,15 +263,43 @@ public void to(MutationComposer composer)
     }
   }
 
-  private boolean isWriteSetOverlappedWith(Scan scan) {
+  /**
+   * Verifies if this scan does not overlap previous writes or deletes using the actual scan result.
+   * Because we support arbitrary conditions in the where clause of a scan (not only ScanAll, but
+   * also Scan and ScanWithIndex), we cannot determine whether the scan results will include a
+   * record whose key is the same as the key specified in the previous writes or deletes, without
+   * knowing the obtained keys in the actual scan. With this check, users can avoid seeing
+   * unexpected scan results that have not included previous writes or deletes yet.
+   *
+   * @param scan the scan to be verified
+   * @param results the results of the scan
+   */
+  public void verify(Scan scan, Map<Snapshot.Key, TransactionResult> results) {
```

Comment:
```suggestion
  public void verifyNoOverlap(Scan scan, Map<Snapshot.Key, TransactionResult> results) {
```
I feel `verify` is probably too general.
Additionally, since the method checks for overlap and throws an exception if one is found, we should specify what is verified in the method name.

Reply:
Fixed in dc79a3fde5a51bf733ad34ca16d120d70f56b720. Thanks!